### DISCUSSION TITLE
Response allowed after certain time limit?

### DISCUSSION THREAD

**Discussion by rachelkytsui** [jspsych/jsPsych#1631] [Q&A]
Hello jsPsych community! (I apologize if this has been asked before.)

I am creating an experiment where participants have to choose between two pictures after listening to an audio stimulus. My question is about allowing a response after certain time limit. I have previously set the parameter "response_allowed_while_playing" parameter to FALSE (using version 6.2.0), such that no response is allowed until the audio finishes. But I would like to change that so that the restriction would be dependent on time and not audio. Is it possible to restrict the response such that a response cannot be made until, for example, 5s has passed while the audio keeps playing? Thank you so much in advance!

**Comment by becky-gilbert** [comment]
Hi @rachelkytsui! I don't think there's an easy way to do that without modifying the plugin file. Usually this is possible by splitting the trial in two, where the first trial had a set duration (e.g. 5s) and no response is allowed, and the second trial allows a response. From the participant's point of view it feels like one trial. But I'm not sure that this would work if you want to play audio at the start of the first trial and switch to the second 'response' trial while the audio is still playing.
Another option is to modify the plugin file. You could add a new parameter for the response delay. Let us know if you want further guidance with that.

**Reply by rachelkytsui** [reply]
Hi @becky-gilbert, thank you for the reply! We did try splitting the trial into two, so that each would play a different audio and response is allowed only in the second phase of the trial. But such a setting is causing some troubles when we tried to run the experiment online via Pavlovia. It seems that browsers are now blocking autoplaying of audios by default such that audios cannot start without a button press. So we wanted to combine the audios into one trial instead of splitting it into two.

I have not tried modifying any plugin file before. Any further guidance on this matter would be greatly appreciated. Thank you so much!

**Reply by becky-gilbert** [reply]
Hi @rachelkytsui! Hmm, the browser autoplay issue should be separate from your decision about whether or not to split the trial in two. As long as you require some kind of user interaction before the first audio trial starts, then that shouldn't be a problem. Note that we're aware of a problem with audio autoplay not working in Safari, even after a button click or key press (see #1445), so be sure to test your experiment in a different browser. And you if you want to allow your participants to use Safari, you can look at the different solutions proposed in the #1445 thread.
